pkgs/development/tools/coder/default.nix

```
@@ -14,18 +24,58 @@ buildGoModule rec {
     hash = "sha256-26RvDJ890MclDB4rtYQ7CcB3NQRXC7sI2cXd689Eq6E=";
   };
 
-  # integration tests require network access
-  doCheck = false;
+  offlineCache = fetchYarnDeps {
+    yarnLock = src + "/site/yarn.lock";
+    hash = "sha256-xWHLllnJOivjxS56Hp9QhV7oNQhW6TROrzIN1B77eKE=";
+  };
+in
+buildGoModule rec {
+  inherit pname version src;
+
+  subPackages = [ "cmd/coder" ];
 
   vendorHash = "sha256-tdqqGM2b8un4BFtvRJsmiIGdb1AOKP8XxcgGg2DilXA=";
 
-  nativeBuildInputs = [ installShellFiles ];
+  ldflags = [
+    "-s"
+    "-w"
+    "-X github.com/coder/coder/buildinfo.tag=${version}"
+  ];
+
+  preBuild = ''
+    export HOME=$TEMPDIR
+
+    cd site
```

nit, consider `pushd` (here and `popd` below) so we do not need to remember exactly where we are

---

Do you mean adding below `cd ..` ?

---

no, sorry should have given a suggestion; something like this:
```suggestion
    pushd site
    
    # do stuff
    
    popd
```

---

Mind blown 😱 . Didn't know that exists 😂